firmware/library/L1_Drivers/uart.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstdint>
+
+class UART_Interface
+{
+ public:
+    virtual void SetBaud(uint32_t baudrate) = 0;
+    virtual bool Initialize(uint32_t baud, uint32_t queue_size = 16) = 0;
+    virtual void InteruptHandler(void) = 0;
```

Comment:
What does this do? I still can't figure out what  UART's would do with this implemented method.

Reply:
This method would be called whenever an interrupt is created due to the Rx pin receiving data. The method would take that data from the pin and throw it onto the queue waiting to be picked up by the user. The user would then use the receive function to grab that data whenever they need to.

Reply:
I dont thing that the interface should include an InterruptHandler. You cannot attach an interrupt to a class's method unless it is static. Therefor, there would need to be a templated static handler for the various UART types. So this method should be removed.